### PR TITLE
Add min-max normalization

### PR DESCRIPTION
**Pull request by djdameln**
# Description
- Adds min-max normalization as an optional method for normalizing the anomaly scores and heatmaps.
- New `MinMaxNormalizationCallback` added
- Add option to change normalization method from the configurable parameters

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: anomalib/core/callbacks/min_max_normalization.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+"""Anomaly Score Normalization Callback that uses min-max normalization."""
```
Comment (ashwinvaidya17) [inline anomalib/core/callbacks/min_max_normalization.py:1]: We will require a license header

File: anomalib/core/callbacks/__init__.py [hunk 2]
```diff
@@ -51,17 +52,22 @@ def get_callbacks(config: Union[ListConfig, DictConfig]) -> List[Callback]:
...
+                raise NotImplementedError("Score Normalization is currently supported for PADIM and STFPM only.")
+        elif config.model.normalization_method == "min_max":
+            callbacks.append(MinMaxNormalizationCallback())
         else:
-            raise NotImplementedError("Score Normalization is currently supported for PADIM and STFPM only.")
+            raise ValueError(f"Normalization method not recognized: {config.model.normalization_method}")
 
     if not config.project.log_images_to == []:
-        callbacks.append(VisualizerCallback(inputs_are_normalized=config.model.normalize_scores))
+        callbacks.append(VisualizerCallback(inputs_are_normalized=config.model.normalization_method == "cdf"))
```
Comment (ashwinvaidya17) [inline anomalib/core/callbacks/__init__.py:70]: Should this be `normalization_method in ("cdf", "min_max")`

File: anomalib/models/padim/config.yaml [hunk 3]
```diff
@@ -35,7 +35,7 @@ model:
 project:
   seed: 42
   path: ./results
-  log_images_to: []
+  log_images_to: [local]
```
Comment (ashwinvaidya17) [inline anomalib/models/padim/config.yaml:38]: Maybe we can keep this disabled.

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks. Looks good. Only a few very minor comments.

File: anomalib/models/padim/config.yaml [hunk 4]
```diff
@@ -3,7 +3,7 @@ dataset:
   format: mvtec
   path: ./datasets/MVTec
   url: ftp://guest:GU.205dldo@ftp.softronics.ch/mvtec_anomaly_detection/mvtec_anomaly_detection.tar.xz
-  category: leather
+  category: bottle
```
Comment (samet-akcay) [inline anomalib/models/padim/config.yaml:6]: Do we need this change? If yes, can we modify the other algos as well?

File: anomalib/core/callbacks/__init__.py [hunk 5]
```diff
@@ -51,17 +52,22 @@ def get_callbacks(config: Union[ListConfig, DictConfig]) -> List[Callback]:
...
 
-    if "normalize_scores" in config.model.keys() and config.model.normalize_scores:
-        if config.model.name in ["padim", "stfpm"]:
-            if not config.optimization.nncf.apply:
-                callbacks.append(AnomalyScoreNormalizationCallback())
+    if "normalization_method" in config.model.keys() and not config.model.normalization_method == "none":
+        if config.model.normalization_method == "cdf":
+            if config.model.name in ["padim", "stfpm"]:
+                if not config.optimization.nncf.apply:
+                    callbacks.append(AnomalyScoreNormalizationCallback())
```
Comment (samet-akcay) [inline anomalib/core/callbacks/__init__.py:59]: I guess this is not super urgent, but we'll have to rename `AnomalyScoreNormalizationCallback` in the future since we now have multiple methods.

File: anomalib/core/model/anomaly_module.py [hunk 6]
```diff
@@ -47,6 +52,7 @@ def __init__(self, params: Union[DictConfig, ListConfig]):
         self.pixel_threshold = AdaptiveThreshold(self.hparams.model.threshold.pixel_default)
 
         self.training_distribution = AnomalyScoreDistribution()
+        self.min_max = MinMax()
```
Comment (samet-akcay) [inline anomalib/core/model/anomaly_module.py:55]: Do we need these two at the same time?

Comment (djdameln) [reply anomalib/core/model/anomaly_module.py:55]: Not really, it depends on the normalization method that's being used. But I figured it doesn't hurt to have it either. Same as we also always have the `pixel_threshold` attribute, even when the task type is classification.

File: anomalib/core/callbacks/normalization.py [hunk 7]
```diff
@@ -2,12 +2,12 @@
...
 
 import pytorch_lightning as pl
-import torch
 from pytorch_lightning import Callback, Trainer
 from pytorch_lightning.utilities.types import STEP_OUTPUT
-from torch.distributions import LogNormal, Normal
+from torch.distributions import LogNormal
 
 from anomalib.models import get_model
+from anomalib.utils.normalization.cdf_normalization import normalize, standardize
```
Comment (samet-akcay) [inline anomalib/core/callbacks/normalization.py:10]: do we need to explicitly name the module as `cdf_normalization`. It's already located in `normalisation` package. I think we could rename it as just `cdf`. in this case, it would be like `from anomalib.utils.normalization.cdf import normalize, standardize`. It would also be slightly less verbose.

File: anomalib/core/callbacks/min_max_normalization.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from typing import Any, Dict
+
+import pytorch_lightning as pl
+from pytorch_lightning import Callback
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+
+from anomalib.utils.normalization.min_max_normalization import normalize
```
Comment (samet-akcay) [inline anomalib/core/callbacks/min_max_normalization.py:23]: Similar to the comment below:
do we need to explicitly name the module as `min_max_normalization`. It's already located in `normalisation` package. I think we could rename it as just `min_max`. in this case, it would be like `from anomalib.utils.normalization.min_max import normalize, standardize`. It would also be slightly less verbose.

**Review by samet-akcay — COMMENTED** [review]
Thanks a lot for the effort @djdameln. Such a great improvement! Only minor comments.

**Review by samet-akcay — APPROVED** [review]
